pkg/lib/accountmanagement/service_identity.go

```diff
@@ -0,0 +1,1333 @@
+		return nil, err
+	}
+
+	return &UpdateIdentityUsernameOutput{IdentityInfo: info}, nil
+}
+
+type DeleteIdentityUsernameInput struct {
+	IdentityID string
+}
+
+type DeleteIdentityUsernameOutput struct {
+	IdentityInfo *identity.Info
+}
+
+func (s *Service) DeleteIdentityUsername(resolvedSession session.ResolvedSession, input *DeleteIdentityUsernameInput) (*DeleteIdentityUsernameOutput, error) {
+	userID := resolvedSession.GetAuthenticationInfo().UserID
+	identityID := input.IdentityID
+
+	var info *identity.Info
+	err := s.Database.WithTx(func() (err error) {
+		info, err = s.prepareDeleteIdentity(userID, identityID)
+		if err != nil {
+			return err
+		}
+
+		err = s.deleteIdentity(info)
+		if err != nil {
+			return err
+		}
+
+		err = s.dispatchIdentityDeletedEvent(info)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &DeleteIdentityUsernameOutput{IdentityInfo: info}, nil
+}
+
+type StartAddIdentityEmailInput struct {
+	LoginID    string
+	LoginIDKey string
+}
+
+type StartAddIdentityEmailOutput struct {
+	IdentityInfo     *identity.Info
+	NeedVerification bool
+	Token            string
+}
+
+func (s *Service) StartAddIdentityEmail(resolvedSession session.ResolvedSession, input *StartAddIdentityEmailInput) (*StartAddIdentityEmailOutput, error) {
+	userID := resolvedSession.GetAuthenticationInfo().UserID
+	loginKey := input.LoginIDKey
+	loginID := input.LoginID
+
+	spec, err := s.makeLoginIDSpec(loginKey, loginID)
+	if err != nil {
+		return nil, err
+	}
+
+	var info *identity.Info
+	var token string
+	var needVerification bool
+	err = s.Database.WithTx(func() error {
+		info, err = s.prepareNewIdentity(userID, spec)
+		if err != nil {
+			return err
+		}
+
+		verified, err := s.checkIdentityVerified(info)
+		if err != nil {
+			return err
+		}
+		needVerification = !verified
+
+		if !verified {
+			return nil
+		}
+
+		err = s.createIdentity(info)
+		if err != nil {
+			return err
+		}
+
+		err = s.dispatchIdentityCreatedEvent(info)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if needVerification {
+		channel, target := info.LoginID.ToChannelTarget()
+		err = s.sendOTPCode(userID, channel, target, false)
+		if err != nil {
+			return nil, err
+		}
+		token, err = s.Store.GenerateToken(GenerateTokenOptions{
+			UserID: userID,
+			Email:  info.LoginID.LoginID,
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &StartAddIdentityEmailOutput{
+		IdentityInfo:     info,
+		NeedVerification: needVerification,
+		Token:            token,
+	}, nil
+}
+
+type ResumeAddIdentityEmailInput struct {
+	LoginIDKey string
+	Token      string
+	Code       string
+}
+
+type ResumeAddIdentityEmailOutput struct {
+	IdentityInfo *identity.Info
+}
+
+func (s *Service) ResumeAddIdentityEmail(resolvedSession session.ResolvedSession, input *ResumeAddIdentityEmailInput) (output *ResumeAddIdentityEmailOutput, err error) {
```

Comment:
I would like to come up with a convention

- If it is a one-step operation, then the signature should be `(resolvedSession, input)`
- If it is the start of a multi-step operation, then it should be `StartXXX(resolvedSession, input)`
- If it is the continuation of a multi-step operation, then it should be `ResumeXXX(resolvedSession, token, input)` or `FinishXXX(resolvedSession, token, input)`